web/src/blueslip.ts

```diff
@@ -184,15 +86,78 @@ export function warn(msg: string, more_info?: unknown): void {
     }
 }
 
-export function error(msg: string, more_info?: unknown, stack = new Error("dummy").stack): void {
+class BlueslipError extends Error {
+    override name = "BlueslipError";
+    more_info?: unknown;
+    constructor(msg: string, more_info?: object | undefined) {
+        super(msg);
+        if (more_info !== undefined) {
+            this.more_info = more_info;
+        }
+    }
+}
+
+export function error(
+    msg: string,
+    more_info?: object | undefined,
+    original_error?: unknown | undefined,
+): void {
+    // original_error could be of any type, because you can "raise"
+    // any type -- something we do see in practice with the error
+    // object being "dead": https://github.com/zulip/zulip/issues/18374
+    let exception: string | Error = msg;
+    if (original_error !== undefined && original_error instanceof Error) {
+        original_error.message = msg;
+        exception = original_error;
+    }
+
+    // Log the Sentry error before the console warning, so we don't
+    // end up with a doubled message in the Sentry logs.
+    Sentry.setContext("more_info", more_info === undefined ? null : more_info);
+    Sentry.getCurrentHub().captureException(exception);
+
     const args = build_arg_list(msg, more_info);
     logger.error(...args);
-    report_error(msg, stack, more_info);
 
+    // Throw an error in development; this will show a dialog (see below).
     if (page_params.development_environment) {
         throw new BlueslipError(msg, more_info);
```

Comment:
I think this class was deleted in this commit?

Reply:
Nope, just moved.  It's now on line 89.